Google.Ads.GoogleAds/examples/AdvancedOperations/AddDemandGenCampaign.cs

```diff
@@ -0,0 +1,485 @@
+                    {
+                        AdGroup = adGroupResourceName,
+                        Ad = ad
+                    }
+                }
+            };
+        }
+        // [END add_demand_gen_campaign_4]
+
+        /// <summary>
+        /// Creates a MutateOperation that creates a new image asset.
+        /// </summary>
+        /// <param name="assetResourceName">The resource name of the text asset to be
+        /// created.</param>
+        /// <param name="url">The url of the image to be retrieved and put into an asset.</param>
+        /// <param name="assetName">The asset name.</param>
+        /// <param name="config">The Google Ads config.</param>
+        /// <returns>A MutateOperation that creates a new image asset.</returns>
+        private MutateOperation CreateImageAssetOperation(
+            string assetResourceName,
+            string url,
+            string assetName,
+            GoogleAdsConfig config)
+            {
+                return new MutateOperation
+                {
+                    AssetOperation = new AssetOperation()
+                    {
+                        Create = new Asset
+                        {
+                            ResourceName = assetResourceName,
+                            ImageAsset = new ImageAsset
+                            {
+                                Data =
+                                ByteString.CopyFrom(
+                                    MediaUtilities.GetAssetDataFromUrl(url, config)
+                                )
+                            },
+                            // Provide a unique friendly name to identify your asset.
+                            // When there is an existing image asset with the same content but a
+                            // different name, the new name will be dropped silently.
+                            Name = assetName
+                        }
+                    }
+                };
+            }
+
+        /// <summary>
+        /// Creates a MutateOperation that creates a new video asset.
+        /// </summary>
+        /// <param name="assetResourceName">The resource name of the text asset to be
+        /// created.</param>
+        /// <param name="videoId">The YouTube id of the video.</param>
+        /// <param name="assetName">The asset name.</param>
+        /// <returns>A MutateOperation that creates a new video asset.</returns>
+        private MutateOperation CreateVideoAssetOperation(
+            string assetResourceName,
+            string videoId,
+            string assetName
+        )
+        {
+            return new MutateOperation
+            {
+                AssetOperation = new AssetOperation
+                {
+                    Create = new Asset
+                    {
+                        ResourceName = assetResourceName,
+                        Name = assetName,
+                        YoutubeVideoAsset = new YoutubeVideoAsset
```

Comment:
Do we need to set the type as `YOUTUBE_VIDEO`?

Reply:
No; at least for .NET, it's inferred by the fact that we're passing a YoutubeVideoAsset, so the created asset will already have type `YOUTUBE_VIDEO`.